### DISCUSSION TITLE
IPFS - file delete behaviour

### DISCUSSION THREAD

**Discussion by criadoperez** [blockstack/gaia#438] [Polls]
The IPFS driver for GAIA will be released soon and there seems to be a discussion on what should happen when files are deleted by a user.

To give some context, IPFS stores files on multiples nodes on the Internet that are not in your control. So if you upload a file, and then try to delete it, they is no guarantee the file really gets deleted everywhere (as you can't force what other nodes do with the files).

For simplicity, the GAIA IPFS driver will get released initially with the simplest solution: The user won't get the option of deleting any files uploaded to IPFS. Question is if we should change this behavior or not. 

A possible solution to this is to cryptographically delete the file. Meaning, a user uploads a file to GAIA like always, but if IPFS is used, GAIA will encrypt the uploaded file before storing it on IPFS (maybe with symmetric encryption, using a different key for every file). GAIA keeps a local copy of these keys. If the file wants to be retrieved, then GAIA will get the encrypted file from IPFS, decrypt it and deliver it to the user decrypted in a transparent way. If the user wants to delete the file, GAIA will simply delete the encryption key. So the encrypted file could still be out there, but no one will be able to use it.

I would like to hear the opinion of the community on this topic. Below are the options currently discussed by the developers.

**Comment by dartman10** [comment]
Will the `update` functionality follow the same decision as for `delete`?  My view is that they are technically the same.  _(I'm assuming the `update` functionality will `delete` the IFPS file then create a new one.)_

**Reply by bjorger** [reply]
Hello Dartman,

The update function works somewhat as intended.
Obviously files cannot be deleted or edited on IPFS, but IPFS has versioning for files, so if you want to update a file, it just shows the recent version.
